# Task 04: Data Storytelling & Statistical Validation

### ApexPlanet Software Pvt. Ltd. - Data Analytics Internship

**Project:** E-commerce Customer Analytics

**Author:** Sunita Satpathy

---

## Objective

The objective of this task is to transform analytical findings into a business story and validate one of the key insights using statistical hypothesis testing. This notebook summarizes insights obtained from previous tasks, presents actionable business recommendations, and performs statistical validation to support data-driven decision making.

In [3]:
import warnings
warnings.filterwarnings("ignore")

import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

from scipy.stats import ttest_ind

plt.style.use("ggplot")

print("Libraries imported successfully.")

Libraries imported successfully.


In [5]:
df = pd.read_csv("../task 02 - EDA and business analytics/cleaned_ecommerce_data.csv")

df.head()

,Order_ID,Order_Date,Customer_ID,Customer_Name,Age,Gender,City,Product,Category,Quantity,Unit_Price,Total_Sales,Calculated_Sales,Month,Age_Group
0,ORD100002,2025-02-25,CUST5529,Customer_227,30.0,Female,Bengaluru,Rice,Grocery,7,2829.77,19808.39,19808.39,February,26-35
1,ORD100003,2025-10-14,CUST3127,Customer_182,63.0,Male,Bengaluru,Book,Education,5,27906.16,139530.80,139530.80,October,56-65
2,ORD100004,2025-05-13,CUST8887,Customer_487,62.0,Female,Bengaluru,Book,Education,8,37491.06,299928.48,299928.48,May,56-65
3,ORD100005,2025-12-02,CUST2515,Customer_470,65.0,Female,Kolkata,Mobile,Electronics,9,28541.36,256872.24,256872.24,December,56-65
4,ORD100006,2025-11-20,CUST4796,Customer_380,44.0,Male,Bengaluru,Rice,Grocery,10,14036.59,140365.90,140365.90,November,36-45


In [6]:
print("Dataset Shape :", df.shape)

print("\nColumns:")
display(df.columns.to_frame(name="Column Name"))

print("\nData Types")
display(df.dtypes.to_frame(name="Data Type"))

print("\nMissing Values")
display(df.isnull().sum().to_frame(name="Missing Values"))

Dataset Shape : (1000, 15)

Columns:


,Column Name
Order_ID,Order_ID
Order_Date,Order_Date
Customer_ID,Customer_ID
Customer_Name,Customer_Name
Age,Age
Gender,Gender
City,City
Product,Product
Category,Category
Quantity,Quantity



Data Types


,Data Type
Order_ID,str
Order_Date,str
Customer_ID,str
Customer_Name,str
Age,float64
Gender,str
City,str
Product,str
Category,str
Quantity,int64



Missing Values


,Missing Values
Order_ID,0
Order_Date,0
Customer_ID,0
Customer_Name,0
Age,0
Gender,0
City,0
Product,0
Category,0
Quantity,0


# Business Objective

The primary objective of this analysis is to convert data insights into meaningful business decisions. Using the cleaned e-commerce dataset, the analysis summarizes customer purchasing behaviour, product performance, demographic trends, and sales patterns.

Finally, a statistical hypothesis test is performed to validate whether the observed sales difference between male and female customers is statistically significant.

# Executive Summary

This project is the final stage of the ApexPlanet Data Analytics Internship. It summarizes the work completed in the previous tasks and presents the overall business findings in a structured manner.

### Previous Work Summary

**Task 01 – Data Immersion & Wrangling**
- Performed data cleaning and preprocessing.
- Handled missing values and inconsistent data.
- Created analysis-ready dataset.

**Task 02 – Exploratory Data Analysis**
- Explored customer purchasing behaviour.
- Identified sales trends across different customer segments.
- Analysed category-wise and city-wise performance.

**Task 03 – Interactive Dashboard**
- Built an interactive Power BI dashboard.
- Designed KPIs for business monitoring.
- Enabled business users to explore sales performance interactively.

This notebook focuses on converting these analytical findings into business insights and validating one important observation using statistical testing.

# Data Storytelling

The e-commerce dataset provides valuable insights into customer purchasing behaviour and overall business performance.

The analysis began with data cleaning to ensure reliable results. Exploratory analysis revealed trends related to customer demographics, product categories, sales performance, and purchasing patterns.

An interactive dashboard was then developed to monitor important KPIs and simplify business decision-making.

Finally, statistical validation is performed to determine whether an observed business pattern is supported by data rather than random variation.

This complete workflow transforms raw business data into actionable insights that can support marketing strategies, customer targeting, and future business planning.

# Key Business Insights

Based on the previous analysis, the following observations were identified:

- Customer purchasing behaviour varies across different demographic groups.
- Certain product categories contribute more to total revenue.
- Sales performance differs across cities.
- The interactive dashboard enables quick monitoring of important business KPIs.
- Statistical validation helps determine whether observed differences are statistically meaningful.

In [7]:
print("="*60)
print("BUSINESS SUMMARY")
print("="*60)

print(f"Total Orders      : {df['Order_ID'].nunique()}")
print(f"Total Customers   : {df['Customer_ID'].nunique()}")
print(f"Total Sales       : ₹{df['Total_Sales'].sum():,.2f}")
print(f"Average Order Value : ₹{df['Total_Sales'].mean():,.2f}")

print("\nTop Category:")
print(df.groupby("Category")["Total_Sales"].sum().sort_values(ascending=False).head(1))

print("\nTop City:")
print(df.groupby("City")["Total_Sales"].sum().sort_values(ascending=False).head(1))

BUSINESS SUMMARY
Total Orders      : 992
Total Customers   : 947
Total Sales       : ₹139,399,439.65
Average Order Value : ₹139,399.44

Top Category:
Category
Electronics    50778581.7
Name: Total_Sales, dtype: float64

Top City:
City
Patna    19285966.89
Name: Total_Sales, dtype: float64


# Statistical Validation

## Hypothesis Testing

To validate one of the business observations, an Independent Two-Sample T-Test is performed.

### Null Hypothesis (H₀)

There is no statistically significant difference in the average sales of male and female customers.

### Alternative Hypothesis (H₁)

There is a statistically significant difference in the average sales of male and female customers.

**Significance Level (α) = 0.05**

In [8]:
male_sales = df[df["Gender"] == "Male"]["Total_Sales"]
female_sales = df[df["Gender"] == "Female"]["Total_Sales"]

print("Male Customers :", len(male_sales))
print("Female Customers :", len(female_sales))

t_stat, p_value = ttest_ind(
    male_sales,
    female_sales,
    equal_var=False
)

print("\nT-statistic :", round(t_stat,4))
print("P-value :", round(p_value,6))

Male Customers : 511
Female Customers : 489

T-statistic : 0.6826
P-value : 0.495011


In [9]:
alpha = 0.05

print("="*60)

if p_value < alpha:
    print("Decision : Reject the Null Hypothesis (H₀)")
    print("\nConclusion:")
    print("There is a statistically significant difference in the average sales of male and female customers.")
else:
    print("Decision : Fail to Reject the Null Hypothesis (H₀)")
    print("\nConclusion:")
    print("There is no statistically significant difference in the average sales of male and female customers.")

print("="*60)

Decision : Fail to Reject the Null Hypothesis (H₀)

Conclusion:
There is no statistically significant difference in the average sales of male and female customers.


# Statistical Interpretation

The Independent Two-Sample T-Test was performed to compare the average sales generated by male and female customers.

### Test Results

- **T-statistic:** 0.6826
- **P-value:** 0.4950
- **Significance Level (α):** 0.05

Since the p-value is greater than the significance level (0.4950 > 0.05), the Null Hypothesis (H₀) cannot be rejected.

### Business Interpretation

The analysis indicates that the observed difference in average sales between male and female customers is not statistically significant. Therefore, gender alone does not appear to be a strong factor influencing customer spending behaviour in this dataset.

This suggests that future marketing strategies should not rely solely on customer gender. Instead, the business should focus on other factors such as product categories, customer age groups, purchasing behaviour, seasonal trends, and city-wise sales performance to improve business outcomes.

# Business Recommendations

Based on the complete analysis, the following recommendations are proposed:

1. Continue monitoring high-performing product categories to maximise revenue.
2. Focus marketing campaigns on customer segments identified through purchasing behaviour rather than gender alone.
3. Develop city-specific marketing strategies to strengthen regional sales.
4. Use interactive dashboards for continuous KPI monitoring and business reporting.
5. Perform regular statistical validation before implementing major business decisions to ensure insights are supported by data.

# Conclusion

This project successfully transformed raw e-commerce data into meaningful business insights through data cleaning, exploratory analysis, dashboard development, and statistical validation.

The statistical hypothesis test showed that there is no significant difference in average sales between male and female customers. This demonstrates the importance of validating analytical observations before making business decisions.

Overall, the project highlights how data analytics can support evidence-based decision-making, improve business understanding, and help organisations develop more effective strategies.